### ISSUE TITLE
Support for additional platforms and architectures

### ISSUE THREAD

**Issue by michael-house** [Unity-Technologies/com.unity.webrtc#15]
As Unity itself is very inclusive of multiple platforms and architectures, ideally this plugin would be similarly inclusive.

Specifically, I'd like to use this package on an IL2CPP UWP ARM build, targeting the HoloLens 2.

I do see work on IL2CPP, but nothing for ARM. It's certainly possible as there are other efforts with this goal: https://webrtc-uwp.github.io/.

**Comment by karasusan** [comment]
I understand your needs.
Our team will discuss about ARM support in the near future.

**Comment by Juanmaramon** [comment]
@rafcsoares Yes, I'm agree that WebGL support can be really helpful for voice/video chat communication for web games/applications. 

Are any efforts in this field?

Thanks!

**Comment by karasusan** [comment]
There is the priority order of the supporting platform. We will develop them in this order.
Do you have any ideas?

1. iOS
2. Android
3. Windows ARM64 (for Hololens2)
4. WebGL

**Comment by michael-house** [comment]
Really any order is great at this point, what's the timeline?

**Comment by nvanofwegen** [comment]
We are really looking forward to WebGL support. We are building teambuilding multiplayer games, where we ideally would have a linux server and WebGL clients connected using WebRTC. 

Currently we have use the WebRTC implementation provided by because-why-not but that one lacks Linux support, so we have to use windows servers. For another game we are trying out Photon, which seems to work but is relatively expensive.

**Comment by ArneBezuijen** [comment]
Agreed, glad WebGL is on the list. A (rough?) time indication along with the priority list would be great. Thanks!

**Comment by dyhmichail** [comment]
> There is the priority order of the supporting platform. We will develop them in this order.
> Do you have any ideas?
> 
> 1. iOS
> 2. Android
> 3. Windows ARM64 (for Hololens2)
> 4. WebGL

I think Android should be prioritized because many other devices like AR Headsets use it.

**Comment by Juanmaramon** [comment]
> 
> 
> There is the priority order of the supporting platform. We will develop them in this order.
> Do you have any ideas?
> 
>     1. iOS
> 
>     2. Android
> 
>     3. Windows ARM64 (for Hololens2)
> 
>     4. WebGL

Well, iOS and Android are multithreaded platforms which let you resolve real time comunications problems with other solutions/protocols. But WebGL is not, this is why webRTC is a key component and play a big deal.

**Comment by karasusan** [comment]
I set milestones which releases every two months.
https://github.com/Unity-Technologies/com.unity.webrtc/milestones?direction=asc&sort=title&state=open

- 2.3 -> iOS
- 2.4 -> Android
- 2.5 -> Windows Arm64

Of course, we do our best, but nobody knows the future. 😅

**Comment by karasusan** [comment]
We are developing additional platform support.
The next one is iOS.

**Comment by foungyu** [comment]
Hi @karasusan , thanks for working on different platforms support, I am looking for video/audio chat between iOS and Hololens 2, and Unity WebRTC is my ideal solution. Any workaround suggestion since Windows ARM64 is not supported yet? Thanks.

**Comment by karasusan** [comment]
@foungyu 
Unfortunately, just now we do not have a workaround to use it for the Windows ARM64 platform.

**Comment by desto12** [comment]
@karasusan is there any chance to support hololens 2 to the end of year? Or can you give some estimated date of this release? 
I would be very grateful for any information :)

**Comment by Thaina** [comment]
Do windows still require NVDIA? And what about just transferring text data? Can this be used on every platform just for data?

**Comment by karasusan** [comment]
@Thaina 

NVIDIA is not required by the WebRTC on Windows.
Transferring text data is supported as `RTCDataChannel`. 
https://docs.unity3d.com/Packages/com.unity.webrtc@2.3/manual/datachannel.html

**Comment by Thaina** [comment]
@karasusan There was said in the limitation though?

https://github.com/Unity-Technologies/com.unity.webrtc#limitations

And how about webgl build of unity? Can it just at least open datachannel with webgl right of now?

**Comment by karasusan** [comment]
@Thaina 

Thanks, This is wrong. I am going to remove this sentence.
And WebGL is not supported, unfortunatelly.

**Comment by Thaina** [comment]
@karasusan Thank you. That was unfortunate for webgl. I wish at least datachannel would be supported on webgl

**Comment by desto12** [comment]
Any news about UWP support? :)

**Comment by karasusan** [comment]
We are pending UWP support because the chromium team is not testing WinRT on arm64 architecture continuously.
https://ci.chromium.org/p/webrtc/g/ci/console

And the second option is using WinRTC but this project looks not active. WinRTC contains libwebrtc, the version of libwebrtc m84 is too old. Currently already shipped m93.
https://github.com/microsoft/winrtc

**Comment by karasusan** [comment]
I noticed this comment that says about the announcement in the README of the MR-WebRTC repo.
The strategy of Microsoft would be allocating resources to Microsoft Mesh instead of supporting WebRTC.
https://github.com/microsoft/MixedReality-WebRTC/issues/414#issuecomment-866596898

**Comment by predictcare** [comment]
Hi @karasusan , sorry but i don't get the connection between Microsoft and the WebRtc implementation for Arm64 Platform. Can you explain us some more ? And, what we, the community can do to help you.
I'm also interested in an Arm64 (or Arm, as well) implementation for Hololens2 of the WebRtc package.

**Comment by karasusan** [comment]
@michael-house
Can we change this issue title to **Windows UWP ARM64 support** ?

**Comment by desto12** [comment]
> @michael-house Can we change this issue title to **Windows UWP ARM64 support** ?

Permission granted :)

**Comment by desto12** [comment]
Hello, is there any news about Holo 2 support?

**Comment by dl4mmers** [comment]
I would also like to share my interest in general UWP support in the future since [Mixed Reality WebRTC](https://github.com/microsoft/MixedReality-WebRTC) isn't devolped anymore and it doesn't support ARM64.

**Comment by kannan-xiao4** [comment]
As mentioned in the thread below, the priority is still low.
https://forum.unity.com/threads/unity-render-streaming-status-update-q4-2021.1215801/

**Comment by karasusan** [comment]
memo: WRS-143

**Comment by desto12** [comment]
Hello and sorry for asking again I know, that it is on roadmap and that it is independent of you, but maybe there are some fresh news?

**Comment by kannan-xiao4** [comment]
@desto12 
The situation has not changed since https://github.com/Unity-Technologies/com.unity.webrtc/issues/15#issuecomment-1082558797.

**Comment by Braineratt** [comment]
Id like to share my interrest in this too, we are currently stuck with [Mixed Reality WebRTC](https://github.com/microsoft/MixedReality-WebRTC) aswell.

I'm guessing there are a lot more developers in this situation, that are more or less waiting to find a replacement.

**Comment by wh569860630** [comment]
When will uwp platform push be supported？？

**Comment by zeedyBOI** [comment]
Is there any plan to support this on near future? Support for Hololens 2 would be great

**Comment by fkajzer** [comment]
also stuck because of dropped [Mixed Reality WebRTC](https://github.com/microsoft/MixedReality-WebRTC)

**Comment by dudziakl** [comment]
I'm also waiting for Unity's support of WebRTC for UWP. As a workaround, you can use any WebRTC website that allows for screen sharing and audio streaming. Open it with Edge browser in the background of your Unity app. You can open the browser from Unity with `Windows.System.Launcher.LaunchUriAsync()` or you could try new WebView2 support to have it embedded https://learn.microsoft.com/en-us/microsoft-edge/webview2/get-started/hololens2 I didn't try it yet but it looks promising. 
I can confirm that I was able to share my video stream from HoloLens which includes holograms through webRTC chat created by Gather (https://app.gather.town/) or Whereby (https://whereby.com/) simply by using the option Share my desktop. Don't share the camera because it will stream RGB camera without holograms. Then you can run any app you like. To close connection you have to get back to the browser and disconnect there. It is not very convenient but is better than nothing.

**Comment by ragavendranbalaCES** [comment]
Hi @dudziakl , I have tried your workaround. but I can't able to share the video with Holograms inside webview for unity. There is no window to select for screensharing.

**Comment by sierpinskid** [comment]
I'm also interested in UWP support. Thanks!
